src/bloom/scalable-bloom-filter.ts

```diff
@@ -0,0 +1,218 @@
+        this._error_rate,
+        this._ratio
+      )
+    )
+    this._filters[this._filters.length - 1].seed = this.seed
+  }
+
+  /**
+   * @override
+   * Return the current seed.
+   * For obscure reason we must code this function...
+   */
+  public get seed() {
+    return this._seed
+  }
+
+  /**
+   * @override
+   * Set the seed for this structure. If you override the seed after adding data,
+   * all the filters will be updated and you may get wrong indexes for already indexed data
+   * due to the seed change. So only change it once before adding data.
+   * @param  seed the new seed that will be used in this structure
+   */
+  public set seed(seed: number) {
+    this._seed = seed
+    this._rng = seedrandom(`${this._seed}`) as prng
+    this._filters.forEach((filter: PartitionBloomFilter) => {
+      filter.seed = this.seed
+    })
+  }
+
+  /**
+   * Add a new element to the filter
+   * @param element
+   */
+  public add(element: HashableInput) {
+    // determine if we need to create a new filter
+    const currentFilter = this._filters[this._filters.length - 1]
+    if (currentFilter._currentload() > currentFilter._loadFactor) {
+      // create a new filter
+      const newSize =
+        this._initial_size *
+        Math.pow(ScalableBloomFilter._s, this._filters.length + 1) *
+        Math.LN2
+      const newErrorRate =
+        this._error_rate * Math.pow(this._ratio, this._filters.length)
+      this._filters.push(
+        PartitionBloomFilter.create(newSize, newErrorRate, this._ratio)
+      )
+      this._filters[this._filters.length - 1].seed = this.seed
+    }
+    // get the newly created filter
+    this._filters[this._filters.length - 1].add(element)
+  }
+
+  /**
+   * Return True if the element has been found, false otherwise.
+   * Check until we found the value in a filter otherwise stop on the first value found.
+   * @param element
+   * @returns
+   */
+  public has(element: HashableInput) {
+    let i = 0
+    let found = false
+    while (!found && i < this._filters.length) {
+      found = this._filters[i].has(element)
+      i++
+    }
+    return found
+  }
+
+  /**
+   * Return the current capacity (number of elements) of this filter
+   * @returns
+   */
+  public capacity(): number {
+    return this._filters.map(f => f._capacity).reduce((p, c) => p + c, 0)
+  }
+
+  /**
+   * Return the current false positive rate of this structure
+   * @returns
+   */
+  public rate(): number {
+    return this._filters[this._filters.length - 1].rate()
+  }
+
+  /**
+   * Check if two ScalableBloomFilter are equal
+   * @param filter
+   * @returns
+   */
+  public equals(filter: ScalableBloomFilter) {
+    // first check the seed
+    if (this.seed !== filter.seed) {
+      return false
+    }
+    // check the ratio
+    if (this._ratio !== filter._ratio) {
+      return false
+    }
+    // check the capacity
+    if (this.capacity() !== filter.capacity()) {
+      return false
+    }
```

Comment:
Why not merge all conditions in the same if, as they all leads to the same return value?
```suggestion
    // assert the seed, the ratio and the capacity are equals
    if (this.seed !== filter.seed || this._ratio !== filter._ratio || this.capacity() !== filter.capacity()) {
      return false
    }
```

Reply:
And yours is less expensive too.